vm_supervisor/metrics.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os
+from os.path import join
+from typing import Optional, Iterable
+from uuid import UUID
+
+from sqlalchemy import Column, Integer, String, Float, DateTime
+from sqlalchemy import create_engine
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+
+from .conf import settings
+
+logger = logging.getLogger(__name__)
+
+
+Base = declarative_base()
+session_maker: Optional[sessionmaker] = None
+
+
+def get_database_sessionmaker() -> sessionmaker:
+    global session_maker
+    if session_maker:
+        return session_maker
+
+    # engine = create_engine('sqlite:///:memory:', echo=True)
+    engine = create_engine(f"sqlite:///{settings.EXECUTION_DATABASE}", echo=True)
+    Base.metadata.create_all(engine)
+    session_maker = sessionmaker(bind=engine)
+    return session_maker
+
+
+class ExecutionRecord(Base):
+    __tablename__ = "records"
+
+    uuid = Column(String, primary_key=True)
```

Comment:
Add `nullable=False` to all columns that cannot be null. The default is `True` in SqlAlchemy iirc.